Align README with docs, clearly state intent and scope of windIO

rafmudaf opened the pull request:
# Improve the language describing windIO, align the README and docs, and fix API docs

This pull request proposes a new intro paragraph to windIO that (I hope) adds clarity around the scope and intent of the project. See the changes to README and the docs homepage.

Additionally, I've copied most of the contents of the docs homepage to the README for consistency. I also deleted the badges since they had broken links. Finally, I added the new `windIO.yaml` module to the autodoc configuration and moved functions that aren't relevant to third-party consumers to private so they are left out of the API docs.

You can see a live version of the [README](https://github.com/rafmudaf/windIO) and [docs](https://rafmudaf.github.io/windIO/) from my fork.

## Related issue
https://github.com/IEAWindSystems/windIO/issues/91

kenloen commented:
> @kenloen I can address [this question](https://github.com/IEAWindSystems/windIO/pull/107#discussion_r2002114071) here if you give your thoughts.

Sorry @rafmudaf, I did not see your comment on the `ruamel.yaml` pull-request - but I am very happy with the changes that do did :+1: It makes I more concise.

rafmudaf commented on README.rst:
```diff
@@ -0,0 +1,73 @@
+The github repository is at https://github.com/IEAWindSystems/windIO and the
+online documentation is at https://ieawindsystems.github.io/windIO.
+
+This windIO repository including the following:
```
Ah thanks, fixed

rafmudaf commented:
@kenloen are you ok with how I've changed the function names in the `windIO.yaml` module? I added a leading underscore to keep them out of the autodocs, but we can probably figure out another way if you'd rather.

kenloen commented:
I am fine with the change to `_get_YAML` . It looks good - thanks :+1:

kenloen commented on windIO/__init__.py:
```diff
@@ -20,7 +20,7 @@
 load_yaml = windIO.yaml.load_yaml
```
In the software lib: https://rafmudaf.github.io/windIO/#module-windIO
It is shown the API would be `windIO.yaml.load_yaml` but I think the preferred way should be `windIO.load_yaml` which is also possible with this line. But it seems that the autodoc extension do not pick that up?!
Should we maybe add that to the `:member:` explicitly?

rafmudaf replied:
Hey @kenloen that's a good point. I tried doing this:
```
.. automodule:: windIO
   :members:

   windIO.validate
   windIO.load_yaml
   windIO.write_yaml
```

but it only lists `windIO.validate`. I agree with your suggestion, but I don't know how to get autodoc to do this.

An alternative is to note the preferred API in the docstring of `windIO.yaml.load_yaml(...)`. Another alternative that maintains an expressive API is to change the function to `windIO.yaml.load(...)` and `windIO.yaml.write(...)` and we use the full path.

kenloen commented:
I have made small changes and I think it now renders as expected?!

I also like your suggestion with the `windIO.yaml.load`, but maybe it could lead to issues if users did `from windIO.yaml import load` if which case it might look as if `load` is a build in function. I therefore think it would be better to maintain `load_yaml`.
